service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ChildController.java

```diff
@@ -62,10 +68,17 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        if (caseData.getChildren1() != null) {
-            caseDetails.getData().put("children1", childrenService.addHiddenValues(caseData));
+        List<Element<Child>> confidentialChildren =
+            confidentialDetailsService.addPartyMarkedConfidentialToList(Child.class, caseData.getAllChildren());
+
+        if (isNotEmpty(confidentialChildren)) {
+            caseDetails.getData().put("confidentialChildren", confidentialChildren);
+        } else {
+            caseDetails.getData().remove("confidentialChildren");
```

Comment:
I think that logic should be implemented somewhere else not in the controller, not sure if the controller should implement logic based on confidentiality.